src/protocol/redpine_cc2500.c

```diff
@@ -0,0 +1,462 @@
+/*
+ This project is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Deviation is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Deviation.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef MODULAR
+  //Allows the linker to properly relocate
+  #define REDPINE_Cmds PROTO_Cmds
+  #pragma long_calls
+#endif
+#include "common.h"
+#include "interface.h"
+#include "mixer.h"
+#include "config/model.h"
+#include "config/tx.h"
+#include "telemetry.h"
+#include "devo.h"
+
+#ifdef MODULAR
+  //Some versions of gcc applythis to definitions, others to calls
+  //So just use long_calls everywhere
+  //#pragma long_calls_off
+  extern unsigned _data_loadaddr;
+  const unsigned long protocol_type = (unsigned long)&_data_loadaddr;
+#endif
+
+#ifdef PROTO_HAS_CC2500
+
+#include "iface_cc2500.h"
+
+
+static const char * const redpine_opts[] = {
+  _tr_noop("AD2GAIN"),  "0", "2000", "655361", NULL,       // big step 10, little step 1
+  _tr_noop("Freq-Fine"),  "-127", "127", NULL,
+  _tr_noop("Format"),  "Fast", "Slow", NULL,
+  _tr_noop("LoopIn.1ms"),  "1", "100", NULL,
+  _tr_noop("PacketSize"),  "14", "100", NULL,
+  NULL
+};
+enum {
+    PROTO_OPTS_AD2GAIN,
+    PROTO_OPTS_FREQFINE,
+    PROTO_OPTS_FORMAT,
+    PROTO_OPTS_LOOPTIME,
+    PROTO_OPTS_PACKETSIZE,
+    LAST_PROTO_OPT,
+};
+ctassert(LAST_PROTO_OPT <= NUM_PROTO_OPTS, too_many_protocol_opts);
+
+#define MAX_PACKET_SIZE 33
+
+// Statics are not initialized on 7e so in initialize() if necessary
+static u8 calData[48][3];
+static u8 channr;
+static u8 ctr;
+static s8 fine;
+static u8 packet_size;
+
+
+// u8 ptr[4] = {0x01,0x12,0x23,0x30};
+//u8 ptr[4] = {0x00,0x11,0x22,0x33};
+static enum {
+  REDPINE_BIND,
+#ifndef EMULATOR
+  REDPINE_BIND_DONE = 1000,
+#else
+  REDPINE_BIND_DONE = 50,
+#endif
+  REDPINE_DATA1,
+} state;
+
+static u16 fixed_id;
+static u8 packet[MAX_PACKET_SIZE];
+
+static const u8 hop_data[] = {
+  0x02, 0xD4, 0xBB, 0xA2, 0x89,
+  0x70, 0x57, 0x3E, 0x25, 0x0C,
+  0xDE, 0xC5, 0xAC, 0x93, 0x7A,
+  0x61, 0x48, 0x2F, 0x16, 0xE8,
+  0xCF, 0xB6, 0x9D, 0x84, 0x6B,
+  0x52, 0x39, 0x20, 0x07, 0xD9,
+  0xC0, 0xA7, 0x8E, 0x75, 0x5C,
+  0x43, 0x2A, 0x11, 0xE3, 0xCA,
+  0xB1, 0x98, 0x7F, 0x66, 0x4D,
+  0x34, 0x1B, 0x00, 0x1D, 0x03
+};
+
+
+static const u16 CRCTable[] = {
+  0x0000,0x1189,0x2312,0x329b,0x4624,0x57ad,0x6536,0x74bf,
```

Comment:
protocol/frskyx_cc2500.c has exact same crc table.

Reply:
Good catch - I'll put that in a shared file